File: bharat-fm/sdks/javascript/src/domains/finance.ts
/**
 * Finance AI domain-specific client for BFMF
 */

import { BharatAIClient } from '../client';
import { GenerationRequest, GenerationResponse } from '../types';
import { BharatAIError } from '../errors';

/**
 * Financial analysis request
 */
export interface FinancialAnalysisRequest {
  /** Financial statements data */
  financialData: {
    balanceSheet?: Record<string, number>;
    incomeStatement?: Record<string, number>;
    cashFlowStatement?: Record<string, number>;
  };
  /** Analysis type */
  analysisType: 'ratio' | 'trend' | 'forecast' | 'valuation' | 'risk';
  /** Time period */
  timePeriod?: {
    start: string;
    end: string;
  };
  /** Industry */
  industry?: string;
  ** Company size */
  companySize?: 'small' | 'medium' | 'large';
  ** Analysis depth */
  analysisDepth?: 'basic' | 'detailed' | 'comprehensive';
}

/**
 * Financial analysis response
 */
export interface FinancialAnalysisResponse {
  /** Analysis ID */
  analysisId: string;
  ** Analysis results */
  analysis: {
    summary: string;
    keyFindings: string[];
    ratios?: Record<string, {
      value: number;
      benchmark?: number;
      status: 'good' | 'average' | 'poor';
    }>;
    trends?: Array<{
      metric: string;
      direction: 'increasing' | 'decreasing' | 'stable';
      change: number;
      significance: 'high' | 'medium' | 'low';
    }>;
    forecasts?: Array<{
      metric: string;
      period: string;
      predicted: number;
      confidence: number;
    }>;
  };
  ** Risk assessment */
  riskAssessment: {
    overallRisk: 'low' | 'medium' | 'high';
    riskFactors: Array<{
      factor: string;
      level: 'low' | 'medium' | 'high';
      impact: 'low' | 'medium' | 'high';
      description: string;
    }>;
  };
  ** Recommendations */
  recommendations: Array<{
    priority: 'high' | 'medium' | 'low';
    action: string;
    expectedImpact: string;
    timeline: string;
  }>;
  /** Analysis time */
  analysisTime: number;
}

/**
 * Transaction audit request
 */
export interface TransactionAuditRequest {
  /** Transaction data */
  transactions: Array<{
    id: string;
    date: string;
    amount: number;
    description: string;
    category: string;
    account: string;
    counterparty?: string;
    reference?: string;
  }>;
  /** Audit type */
  auditType: 'fraud' | 'compliance' | 'error' | 'anomaly';
  /** Risk threshold */
  riskThreshold?: number;
  ** Focus areas */
  focusAreas?: string[];
  ** Time period */
  timePeriod?: {
    start: string;
    end: string;
  };
  ** Account types */
  accountTypes?: string[];
}

/**
 * Transaction audit response
 */
export interface TransactionAuditResponse {
  /** Audit ID */
  auditId: string;
  ** Summary */
  summary: {
    totalTransactions: number;
    flaggedTransactions: number;
    riskScore: number;
    auditPeriod: string;
  };
  ** Flagged transactions */
  flaggedTransactions: Array<{
    transactionId: string;
    riskLevel: 'low' | 'medium' | 'high' | 'critical';
    riskScore: number;
    flags: string[];
    explanation: string;
    recommendedAction: string;
  }>;
  ** Patterns detected */
  patterns: Array<{
    pattern: string;
    frequency: number;
    riskLevel: 'low' | 'medium' | 'high';
    description: string;
  }>;
  ** Compliance status */
  complianceStatus: {
    overall: 'compliant' | 'non-compliant' | 'partial';
    issues: Array<{
      regulation: string;
      status: 'compliant' | 'non-compliant';
      description: string;
      severity: 'low' | 'medium' | 'high';
    }>;
  };
  /** Recommendations */
  recommendations: Array<{
    category: 'process' | 'system' | 'training' | 'policy';
    recommendation: string;
    priority: 'high' | 'medium' | 'low';
  }>;
  /** Audit time */
  auditTime: number;
}

/**
 * Risk assessment request
 */
export interface RiskAssessmentRequest {
  /** Entity information */
  entity: {
    type: 'individual' | 'business' | 'investment';
    name: string;
    industry?: string;
    size?: string;
    location?: string;
  };
  /** Assessment type */
  assessmentType: 'credit' | 'market' | 'operational' | 'liquidity' | 'compliance';
  /** Time horizon */
  timeHorizon: 'short-term' | 'medium-term' | 'long-term';
  /** Risk factors to consider */
  riskFactors?: string[];
  ** Historical data */
  historicalData?: Record<string, any>;
}

/**
 * Risk assessment response
 */
export interface RiskAssessmentResponse {
  /** Assessment ID */
  assessmentId: string;
  ** Overall risk rating */
  overallRiskRating: {
    score: number; // 0-100
    level: 'low' | 'medium' | 'high' | 'critical';
    confidence: number; // 0-100
  };
  ** Risk breakdown */
  riskBreakdown: Array<{
    category: string;
    score: number;
    level: 'low' | 'medium' | 'high' | 'critical';
    factors: Array<{
      factor: string;
      impact: number;
      likelihood: number;
      description: string;
    }>;
  }>;
  ** Key risk drivers */
  keyRiskDrivers: Array<{
    driver: string;
    impact: 'high' | 'medium' | 'low';
    trend: 'increasing' | 'stable' | 'decreasing';
    description: string;
  }>;
  ** Mitigation strategies */
  mitigationStrategies: Array<{
    risk: string;
    strategy: string;
    effectiveness: 'high' | 'medium' | 'low';
      cost: 'low' | 'medium' | 'high';
      timeline: string;
  }>;
  ** Monitoring recommendations */
  monitoringRecommendations: Array<{
    metric: string;
    frequency: string;
    threshold: number;
    action: string;
  }>;
  /** Assessment time */
  assessmentTime: number;
}

/**
 * Market prediction request
 */
export interface MarketPredictionRequest {
  /** Market/Instrument */
  market: {
    type: 'stock' | 'bond' | 'commodity' | 'currency' | 'index';
    symbol: string;
    name: string;
  };
  /** Prediction type */
  predictionType: 'price' | 'trend' | 'volatility' | 'correlation';
  /** Time horizon */
  timeHorizon: '1d' | '1w' | '1m' | '3m' | '6m' | '1y';
  /** Confidence level */
  confidenceLevel?: number;
  /** Include factors */
  includeFactors?: string[];
}

/**
 * Market prediction response
 */
export interface MarketPredictionResponse {
  /** Prediction ID */
  predictionId: string;
  ** Market info */
  market: {
    type: string;
    symbol: string;
    name: string;
    currentPrice?: number;
  };
  ** Predictions */
  predictions: Array<{
    type: string;
    horizon: string;
    prediction: any;
    confidence: number;
    factors: Array<{
      factor: string;
      impact: 'positive' | 'negative';
      weight: number;
    }>;
  }>;
  ** Technical indicators */
  technicalIndicators?: Record<string, {
    value: number;
    signal: 'buy' | 'sell' | 'hold';
    strength: 'weak' | 'moderate' | 'strong';
  }>;
  ** Risk factors */
  riskFactors: Array<{
    factor: string;
    level: 'low' | 'medium' | 'high';
    description: string;
  }>;
  ** Disclaimer */
  disclaimer: string;
  /** Prediction time */
  predictionTime: number;
}

/**
 * Finance AI domain-specific client
 */
export class FinanceAIClient {
  private client: BharatAIClient;
  private domainEndpoint: string;

  /**
   * Create a new Finance AI client
   */
  constructor(client: BharatAIClient, domainEndpoint: string = '/finance') {
    this.client = client;
    this.domainEndpoint = domainEndpoint;
  }

  /**
   * Analyze financial data
   */
  async analyzeFinancials(request: FinancialAnalysisRequest): Promise<FinancialAnalysisResponse> {
    const response = await this.client['makeRequest']<FinancialAnalysisResponse>({
      method: 'POST',
      url: `${this.domainEndpoint}/analyze-financials`,
      data: {
        financial_data: request.financialData,
        analysis_type: request.analysisType,
        time_period: request.timePeriod,
        industry: request.industry,
        company_size: request.companySize,
        analysis_depth: request.analysisDepth
      }
    });

    return response;
  }

  /**
   * Audit transactions
   */
  async auditTransactions(request: TransactionAuditRequest): Promise<TransactionAuditResponse> {
    const response = await this.client['makeRequest']<TransactionAuditResponse>({
      method: 'POST',
      url: `${this.domainEndpoint}/audit-transactions`,
      data: {
        transactions: request.transactions,
        audit_type: request.auditType,
        risk_threshold: request.riskThreshold,
        focus_areas: request.focusAreas,
        time_period: request.timePeriod,
        account_types: request.accountTypes
      }
    });

    return response;
  }

  /**
   * Assess risk
   */
  async assessRisk(request: RiskAssessmentRequest): Promise<RiskAssessmentResponse> {
    const response = await this.client['makeRequest']<RiskAssessmentResponse>({
      method: 'POST',
      url: `${this.domainEndpoint}/assess-risk`,
      data: {
        entity: request.entity,
        assessment_type: request.assessmentType,
        time_horizon: request.timeHorizon,
        risk_factors: request.riskFactors,
        historical_data: request.historicalData
      }
    });

    return response;
  }

  /**
   * Predict market movements
   */
  async predictMarket(request: MarketPredictionRequest): Promise<MarketPredictionResponse> {
    const response = await this.client['makeRequest']<MarketPredictionResponse>({
      method: 'POST',
      url: `${this.domainEndpoint}/predict-market`,
      data: {
        market: request.market,
        prediction_type: request.predictionType,
        time_horizon: request.timeHorizon,
        confidence_level: request.confidenceLevel,
        include_factors: request.includeFactors
      }
    });

    return response;
  }

  /**
   * Generate financial report
   */
  async generateFinancialReport(params: {
    reportType: 'annual' | 'quarterly' | 'monthly' | 'custom';
    financialData: any;
    includeCharts?: boolean;
    includeRatios?: boolean;
    includeForecast?: boolean;
    language?: string;
  }): Promise<{
    reportId: string;
    report: string;
    sections: Array<{
      title: string;
      content: string;
      charts?: Array<{
        type: string;
        title: string;
        data: any;
      }>;
    }>;
    summary: string;
    generatedAt: string;
  }> {
    const response = await this.client['makeRequest']<{
      reportId: string;
      report: string;
      sections: Array<{
        title: string;
        content: string;
        charts?: Array<{
          type: string;
          title: string;
          data: any;
        }>;
      }>;
      summary: string;
      generatedAt: string;
    }>({
      method: 'POST',
      url: `${this.domainEndpoint}/generate-report`,
      data: params
    });

    return response;
  }

  /**
   * Monitor transactions in real-time
   */
  async monitorTransactions(params: {
    accounts: string[];
    rules: Array<{
      condition: string;
      action: 'alert' | 'block' | 'review';
      threshold?: number;
    }>;
    webhookUrl?: string;
  }): Promise<{
    monitoringId: string;
    status: 'active' | 'paused' | 'stopped';
    accounts: string[];
    rules: Array<{
      condition: string;
      action: string;
      threshold?: number;
      triggered: number;
    }>;
    createdAt: string;
  }> {
    const response = await this.client['makeRequest']<{
      monitoringId: string;
      status: 'active' | 'paused' | 'stopped';
      accounts: string[];
      rules: Array<{
        condition: string;
        action: string;
        threshold?: number;
        triggered: number;
      }>;
      createdAt: string;
    }>({
      method: 'POST',
      url: `${this.domainEndpoint}/monitor-transactions`,
      data: params
    });

    return response;
  }

  /**
   * Get compliance requirements
   */
  async getComplianceRequirements(industry: string, jurisdiction: string): Promise<Array<{
    regulation: string;
    description: string;
    requirements: string[];
    penalties: string[];
    frequency: string;
    lastUpdated: string;
  }>> {
    const response = await this.client['makeRequest']<Array<{
      regulation: string;
      description: string;
      requirements: string[];
      penalties: string[];
      frequency: string;
      lastUpdated: string;
    }>>({
      method: 'GET',
      url: `${this.domainEndpoint}/compliance-requirements`,
      params: { industry, jurisdiction }
    });

    return response;
  }

  /**
   * Calculate financial ratios
   */
  async calculateRatios(financialData: any): Promise<Record<string, {
    value: number;
    formula: string;
    interpretation: string;
    benchmark?: {
      industry: string;
      value: number;
    };
  }>> {
    const response = await this.client['makeRequest']<Record<string, {
      value: number;
      formula: string;
      interpretation: string;
      benchmark?: {
        industry: string;
        value: number;
      };
    }>>({
      method: 'POST',
      url: `${this.domainEndpoint}/calculate-ratios`,
      data: { financial_data: financialData }
    });

    return response;
  }

  /**
   * Generate investment recommendation
   */
  async generateInvestmentRecommendation(params: {
    riskProfile: 'conservative' | 'moderate' | 'aggressive';
    investmentAmount: number;
    timeHorizon: string;
    goals: string[];
    currentHoldings?: Array<{
      asset: string;
      value: number;
      allocation: number;
    }>;
  }): Promise<{
    recommendationId: string;
    riskProfile: string;
    recommendedAllocation: Array<{
      assetClass: string;
      percentage: number;
      amount: number;
      rationale: string;
    }>;
    specificInvestments: Array<{
      name: string;
      type: string;
      allocation: number;
      expectedReturn: number;
      riskLevel: string;
    }>;
    expectedReturns: {
      annual: number;
      timeframe: string;
    };
    risks: Array<{
      type: string;
      level: string;
      mitigation: string;
    }>;
    nextReviewDate: string;
  }> {
    const response = await this.client['makeRequest']<{
      recommendationId: string;
      riskProfile: string;
      recommendedAllocation: Array<{
        assetClass: string;
        percentage: number;
        amount: number;
        rationale: string;
      }>;
      specificInvestments: Array<{
        name: string;
        type: string;
        allocation: number;
        expectedReturn: number;
        riskLevel: string;
      }>;
      expectedReturns: {
        annual: number;
        timeframe: string;
      };
      risks: Array<{
        type: string;
        level: string;
        mitigation: string;
      }>;
      nextReviewDate: string;
    }>({
      method: 'POST',
      url: `${this.domainEndpoint}/investment-recommendation`,
      data: params
    });

    return response;
  }
}

File: bharat-fm/sdks/go/internal/domains/finance.go
package domains

import (
    "context"
    "fmt"

    "github.com/bharat-ai/bharat-fm/sdks/go/internal/client"
    "github.com/bharat-ai/bharat-fm/sdks/go/internal/models"
)

// FinanceAIClient provides finance-specific AI capabilities
type FinanceAIClient struct {
    client *client.Client
}

// NewFinanceAIClient creates a new finance AI client
func NewFinanceAIClient(client *client.Client) *FinanceAIClient {
    return &FinanceAIClient{
        client: client,
    }
}

// FinancialAnalysisRequest represents a financial analysis request
type FinancialAnalysisRequest struct {
    FinancialData map[string]interface{} `json:"financial_data"`
    AnalysisType  string                 `json:"analysis_type"`
    TimePeriod    *struct {
        Start string `json:"start"`
        End   string `json:"end"`
    } `json:"time_period,omitempty"`
    Industry      string `json:"industry,omitempty"`
    CompanySize   string `json:"company_size,omitempty"`
    AnalysisDepth string `json:"analysis_depth,omitempty"`
}

// FinancialAnalysisResponse represents a financial analysis response
type FinancialAnalysisResponse struct {
    AnalysisID string `json:"analysis_id"`
    Analysis   struct {
        Summary      string `json:"summary"`
        KeyFindings  []string `json:"key_findings"`
        Ratios       map[string]struct {
            Value      float64 `json:"value"`
            Benchmark  float64 `json:"benchmark,omitempty"`
            Status     string   `json:"status"`
        } `json:"ratios,omitempty"`
        Trends []struct {
            Metric      string  `json:"metric"`
            Direction   string  `json:"direction"`
            Change      float64 `json:"change"`
            Significance string  `json:"significance"`
        } `json:"trends,omitempty"`
        Forecasts []struct {
            Metric     string  `json:"metric"`
            Period     string  `json:"period"`
            Predicted  float64 `json:"predicted"`
            Confidence float64 `json:"confidence"`
        } `json:"forecasts,omitempty"`
    } `json:"analysis"`
    RiskAssessment struct {
        OverallRisk   string `json:"overall_risk"`
        RiskFactors   []struct {
            Factor     string `json:"factor"`
            Level      string `json:"level"`
            Impact     string `json:"impact"`
            Description string `json:"description"`
        } `json:"risk_factors"`
    } `json:"risk_assessment"`
    Recommendations []struct {
        Priority      string `json:"priority"`
        Action        string `json:"action"`
        ExpectedImpact string `json:"expected_impact"`
        Timeline      string `json:"timeline"`
    } `json:"recommendations"`
    AnalysisTime float64 `json:"analysis_time"`
}

// TransactionAuditRequest represents a transaction audit request
type TransactionAuditRequest struct {
    Transactions []struct {
        ID           string  `json:"id"`
        Date         string  `json:"date"`
        Amount       float64 `json:"amount"`
        Description  string  `json:"description"`
        Category     string  `json:"category"`
        Account      string  `json:"account"`
        Counterparty string  `json:"counterparty,omitempty"`
        Reference    string  `json:"reference,omitempty"`
    } `json:"transactions"`
    AuditType    string `json:"audit_type"`
    RiskThreshold float64 `json:"risk_threshold,omitempty"`
    FocusAreas   []string `json:"focus_areas,omitempty"`
    TimePeriod   *struct {
        Start string `json:"start"`
        End   string `json:"end"`
    } `json:"time_period,omitempty"`
    AccountTypes []string `json:"account_types,omitempty"`
}

// TransactionAuditResponse represents a transaction audit response
type TransactionAuditResponse struct {
    AuditID string `json:"audit_id"`
    Summary struct {
        TotalTransactions    int     `json:"total_transactions"`
        FlaggedTransactions int     `json:"flagged_transactions"`
        RiskScore           float64 `json:"risk_score"`
        AuditPeriod         string  `json:"audit_period"`
    } `json:"summary"`
    FlaggedTransactions []struct {
        TransactionID string `json:"transaction_id"`
        RiskLevel      string `json:"risk_level"`
        RiskScore      float64 `json:"risk_score"`
        Flags          []string `json:"flags"`
        Explanation    string `json:"explanation"`
        RecommendedAction string `json:"recommended_action"`
    } `json:"flagged_transactions"`
    Patterns []struct {
        Pattern     string  `json:"pattern"`
        Frequency   int     `json:"frequency"`
        RiskLevel   string  `json:"risk_level"`
        Description string  `json:"description"`
    } `json:"patterns"`
    ComplianceStatus struct {
        Overall string `json:"overall"`
        Issues []struct {
            Regulation string `json:"regulation"`
            Status     string `json:"status"`
            Description string `json:"description"`
            Severity    string `json:"severity"`
        } `json:"issues"`
    } `json:"compliance_status"`
    Recommendations []struct {
        Category      string `json:"category"`
        Recommendation string `json:"recommendation"`
        Priority      string `json:"priority"`
    } `json:"recommendations"`
    AuditTime float64 `json:"audit_time"`
}

// AnalyzeFinancials analyzes financial data
func (c *FinanceAIClient) AnalyzeFinancials(ctx context.Context, req *FinancialAnalysisRequest) (*FinancialAnalysisResponse, error) {
    var result FinancialAnalysisResponse
    
    apiReq := map[string]interface{}{
        "financial_data": req.FinancialData,
        "analysis_type": req.AnalysisType,
    }
    
    if req.TimePeriod != nil {
        apiReq["time_period"] = req.TimePeriod
    }
    if req.Industry != "" {
        apiReq["industry"] = req.Industry
    }
    if req.CompanySize != "" {
        apiReq["company_size"] = req.CompanySize
    }
    if req.AnalysisDepth != "" {
        apiReq["analysis_depth"] = req.AnalysisDepth
    }

    resp, err := c.client.GetHttpClient().R().
        SetContext(ctx).
        SetBody(apiReq).
        SetResult(&result).
        Post("/finance/analyze-financials")
    
    if err != nil {
        return nil, fmt.Errorf("financial analysis request failed: %w", err)
    }
    
    if resp.StatusCode() != 200 {
        return nil, fmt.Errorf("financial analysis API error: status %d", resp.StatusCode())
    }
    
    return &result, nil
}

// AuditTransactions audits transactions
func (c *FinanceAIClient) AuditTransactions(ctx context.Context, req *TransactionAuditRequest) (*TransactionAuditResponse, error) {
    var result TransactionAuditResponse
    
    apiReq := map[string]interface{}{
        "transactions": req.Transactions,
        "audit_type":   req.AuditType,
    }
    
    if req.RiskThreshold > 0 {
        apiReq["risk_threshold"] = req.RiskThreshold
    }
    if len(req.FocusAreas) > 0 {
        apiReq["focus_areas"] = req.FocusAreas
    }
    if req.TimePeriod != nil {
        apiReq["time_period"] = req.TimePeriod
    }
    if len(req.AccountTypes) > 0 {
        apiReq["account_types"] = req.AccountTypes
    }

    resp, err := c.client.GetHttpClient().R().
        SetContext(ctx).
        SetBody(apiReq).
        SetResult(&result).
        Post("/finance/audit-transactions")
    
    if err != nil {
        return nil, fmt.Errorf("transaction audit request failed: %w", err)
    }
    
    if resp.StatusCode() != 200 {
        return nil, fmt.Errorf("transaction audit API error: status %d", resp.StatusCode())
    }
    
    return &result, nil
}

// GetComplianceRequirements gets compliance requirements
func (c *FinanceAIClient) GetComplianceRequirements(ctx context.Context, industry, jurisdiction string) ([]struct {
    Regulation   string   `json:"regulation"`
    Description  string   `json:"description"`
    Requirements []string `json:"requirements"`
    Penalties    []string `json:"penalties"`
    Frequency    string   `json:"frequency"`
    LastUpdated  string   `json:"last_updated"`
}, error) {
    var result []struct {
        Regulation   string   `json:"regulation"`
        Description  string   `json:"description"`
        Requirements []string `json:"requirements"`
        Penalties    []string `json:"penalties"`
        Frequency    string   `json:"frequency"`
        LastUpdated  string   `json:"last_updated"`
    }

    resp, err := c.client.GetHttpClient().R().
        SetContext(ctx).
        SetResult(&result).
        Get("/finance/compliance-requirements").
        SetQueryParams(map[string]string{
            "industry":     industry,
            "jurisdiction": jurisdiction,
        })
    
    if err != nil {
        return nil, fmt.Errorf("get compliance requirements request failed: %w", err)
    }
    
    if resp.StatusCode() != 200 {
        return nil, fmt.Errorf("get compliance requirements API error: status %d", resp.StatusCode())
    }
    
    return result, nil
}

// SimpleAnalyzeFinancials provides a simple financial analysis interface
func (c *FinanceAIClient) SimpleAnalyzeFinancials(ctx context.Context, financialData map[string]interface{}, analysisType string) (string, error) {
    req := &FinancialAnalysisRequest{
        FinancialData: financialData,
        AnalysisType:  analysisType,
    }
    
    resp, err := c.AnalyzeFinancials(ctx, req)
    if err != nil {
        return "", err
    }
    
    return resp.Analysis.Summary, nil
}

// SimpleAuditTransactions provides a simple transaction audit interface
func (c *FinanceAIClient) SimpleAuditTransactions(ctx context.Context, transactions []map[string]interface{}, auditType string) (string, error) {
    // Convert transactions to the required format
    txnList := make([]struct {
        ID           string  `json:"id"`
        Date         string  `json:"date"`
        Amount       float64 `json:"amount"`
        Description  string  `json:"description"`
        Category     string  `json:"category"`
        Account      string  `json:"account"`
        Counterparty string  `json:"counterparty,omitempty"`
        Reference    string  `json:"reference,omitempty"`
    }, len(transactions))
    
    for i, txn := range transactions {
        txnList[i] = struct {
            ID           string  `json:"id"`
            Date         string  `json:"date"`
            Amount       float64 `json:"amount"`
            Description  string  `json:"description"`
            Category     string  `json:"category"`
            Account      string  `json:"account"`
            Counterparty string  `json:"counterparty,omitempty"`
            Reference    string  `json:"reference,omitempty"`
        }{
            ID:           getString(txn, "id"),
            Date:         getString(txn, "date"),
            Amount:       getFloat64(txn, "amount"),
            Description:  getString(txn, "description"),
            Category:     getString(txn, "category"),
            Account:      getString(txn, "account"),
            Counterparty: getString(txn, "counterparty"),
            Reference:    getString(txn, "reference"),
        }
    }
    
    req := &TransactionAuditRequest{
        Transactions: txnList,
        AuditType:    auditType,
    }
    
    resp, err := c.AuditTransactions(ctx, req)
    if err != nil {
        return "", err
    }
    
    return fmt.Sprintf("Audit completed. Risk score: %.2f, Flagged transactions: %d", 
        resp.Summary.RiskScore, resp.Summary.FlaggedTransactions), nil
}

// Helper functions for simple interface
func getString(m map[string]interface{}, key string) string {
    if val, ok := m[key]; ok {
        if str, ok := val.(string); ok {
            return str
        }
    }
    return ""
}

func getFloat64(m map[string]interface{}, key string) float64 {
    if val, ok := m[key]; ok {
        if num, ok := val.(float64); ok {
            return num
        }
        if num, ok := val.(int); ok {
            return float64(num)
        }
    }
    return 0
}